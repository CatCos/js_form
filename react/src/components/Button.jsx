import React, { PropTypes } from 'react';

/**
 * Render function
 *
 * @params props Component properties
 */
function Button(props) {
  return (
    <div className="form-group">
        <button className="submit-button" onClick={props.onSubmit}>{props.text}</button>
    </div>
  );
}

export default Button;
