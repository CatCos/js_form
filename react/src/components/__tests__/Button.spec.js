import React from 'react';
import Button from '../Button';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <Button
      className="submit-button"
      onClick={jest.fn()}
      text='Submit comment'/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
