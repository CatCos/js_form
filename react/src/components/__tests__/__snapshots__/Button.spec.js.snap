// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="form-group"
>
  <button
    className="submit-button"
    onClick={undefined}
  >
    Submit comment
  </button>
</div>
`;
